### PR TITLE
Reader: Create new site header for the reader

### LINKED ISSUE #21480: Reader: Implement new design for site header
Implement new header design for selected sites on Reader.

Design specs: pcdRpT-3xc-p2

### PR DESCRIPTION
**Pull request by wargcm**
Closes #21480 

## Description

- Adds the new site header UI
- Fixes an issue with the reader header going into safe areas

@osullivanchris I limited the site's description to 3 lines, the screenshots below show what it looks like. The previous header did not have any limits for this, and I found that you can set a very long description 😅. Please let me know if that's alright or if you want me to change it to something else.

<details><summary>Long site description example without limit</summary>

> **Note**
> This is what I actually set the site description to, it isn't test data.

Simulator Screenshot - iPhone 14 Pro - 2023-08-31 at 23 04 40

</details>

## Screenshots

| Light | Dark |
|------|------|
| light | dark |

## Testing

To test:
- Launch the app and login
- Enable the feature flag `Reader Improvements v1`
- Go to the reader
- On any card, tap on the site
- 🔎 **Verify** the new design is shown
- 🔎 **Verify** the follow/unfollow button works
- Navigate back
- Open various sites and 🔎 **verify** the post and follower counts are formatted correctly

## Regression Notes
1. Potential unintended areas of impact
Current reader stream headers

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Review by osullivanchris — APPROVED** [review]
Looks great. Two small things that appear per design, but we might tweak:

- the follow button looks a bit too close to the number of followers/posts. I just noticed in Figma the vertical padding between was 12px, should be 16px. Assuming you followed 12 and thats why it looks a bit tight.
- the style of the followers/posts metadata makes me expect I can tap on the number, but I can't. I think if we change the number from semibold to regular, it will be better. 


re; the long description. Capping to 3 lines initially makes sense. But does mean there's no way to read the full description if the reader wants to. Would there be an easy way to display the full description? e.g. tapping the 3 lines expands the area?

File: WordPress/Classes/ViewRelated/Reader/ReaderSiteHeaderView.swift [hunk 1]
```diff
@@ -0,0 +1,217 @@
+import SwiftUI
+
+class ReaderSiteHeaderView: UIView, ReaderStreamHeader {
```
Comment (dvdchr) [inline WordPress/Classes/ViewRelated/Reader/ReaderSiteHeaderView.swift:3]: Good idea! I think I can follow this later when we're adding metadata for the Tag feed page.

**Review by dvdchr — APPROVED** [review]
Looks good! :shipit:

**Comment by wargcm** [comment]
@osullivanchris I increased the spacing for the button to 16px and changed the posts/followers to regular in https://github.com/wordpress-mobile/WordPress-iOS/pull/21482/commits/3631fcd4d7a2c5579d6015fac28afdd6b50265b4.

Here is a screenshot with the new changes:
<details><summary>Updated UI screenshot</summary>

Screenshot

</details>

@dvdchr I'm going to leave this PR open until next week or the beginning of the following week since I made some changes that affect all the reader headers.

**Comment by osullivanchris** [comment]
@wargcm looks great, thanks for the tweaks!

**Comment by dvdchr** [comment]
Hey @wargcm. I added back the Share button via https://github.com/wordpress-mobile/WordPress-iOS/pull/21482/commits/40d1e4db112f95878a659dbc4aae5909c67c938b as per the discussion in p1694597238811179-slack-C05N140C8H5

**Pull request merged**